static-instrumenter/agent-instrumenter/src/main/java/io/opentelemetry/contrib/staticinstrumenter/Main.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.lang.instrument.ClassFileTransformer;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.jar.JarFile;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Main {
+
+  private static final Logger logger = LoggerFactory.getLogger(Main.class);
+
+  private static final Main INSTANCE = new Main(new ClassArchive.Factory() {});
+
+  private final ClassArchive.Factory classArchiveFactory;
+
+  private final Map<String, byte[]> instrumentedClasses = new ConcurrentHashMap<>();
+  private final Map<String, byte[]> additionalClasses = new ConcurrentHashMap<>();
+
+  @SuppressWarnings("ThreadLocalUsage")
```

Comment:
Think we may as well just make it static

Reply:
It's tied to singleton, so implicitely static. I added the suppression as the tool is not smart enough to catch that.

Reply:
Yeah since it's effectively static by being a singleton we may as well declare it static and pass the tool right? There's a reasonable chance of changing the dependency injection strategy in the future while no chance of needing a non-static threadlocal regardless - the tool wins.

Reply:
You lost me here. If we make the currentClass static we need to resign from singleton entirely and make whole code static, right?

Reply:
it's rare that ThreadLocals need to be instance fields (thus the errorprone warning) since they are really tied to the thread

the only reason you would need it to be instance field is if you needed to track multiple "current" classes at the same time across different instances of the containing class

Reply:
Singleton was used only to avoid declaring all methods static and using Main in a purely procedural style. From runtime point of view I believe these are equal.

Reply:
They should be basically same at runtime so we'd like to use static here to reduce exceptions from the tooling

Reply:
Right, can change that, although from usage point of view (see PostTransformer) it will look strange:
- get class using static call to Main
- add it to the list kept by Main singleton (instance call)
In my opinion as from runtime point of view there are no differences (singleton vs static) readability wins, so it's better to keep all Main methods instance. Yet, if you feel strong about making field and access methods to currentClass static, will do that (just say that you do ;) ).

Reply:
Done then :)